pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -100,6 +101,34 @@ type InfoItem struct {
 
 	// Value is human readable content.
 	Value string `json:"value,omitempty"`
+
+	// Value is derived from another source.
+	ValueFrom *InfoItemSource `json:"valueFrom,omitempty"`
+}
+
+type InfoItemSource struct {
+	SecretKeyRef    *core.SecretKeySelector    `json:"secretKeyRef,omitempty"`
+	ConfigMapKeyRef *core.ConfigMapKeySelector `json:"configMapKeyRef,omitempty"`
+	ServiceRef      *ServiceSelector           `json:"serviceRef,omitempty"`
+	IngressRef      *IngressSelector           `json:"ingressRef,omitempty"`
+}
+
+type ServiceSelector struct {
+	// The Service to select from.
+	core.LocalObjectReference `json:",inline"`
```

Comment:
If you use LocalObjectReference the Service must appear in the same namespace as the Application. The same is true of Ingress. Is this only meant to work with load balanced services?

Reply:
Changed to `core.ObjectReference`. Discovered that `core.ConfigMapKeySelector` and such also use `LocalObjectReference`, so I created new `ConfigMapKeySelector` and `SecretKeySelector`.